utils/user_management.py

```diff
@@ -0,0 +1,158 @@
+'''
+Utility for managing user names and passwords for TabPy.
+'''
+
+from argparse import ArgumentParser
+import logging
+import os
+import random
+import sys
+from tabpy_server.app.util import parse_pwd_file
+from tabpy_server.handlers.util import hash_password
+
+logger = logging.getLogger(__name__)
+
+
+def build_cli_parser():
+    parser = ArgumentParser(
+        description=__doc__,
+        epilog='''
+            For more information about how to configure and
+            use authentication for TabPy read documentation
+            at https://github.com/tableau/TabPy
+            ''',
+        argument_default=None,
+        add_help=True)
+    parser.add_argument(
+        'command',
+        choices=['add', 'update'],
+        help='Command to execute')
+    parser.add_argument(
+        '-u',
+        '--username',
+        help='Username to add to passwords file')
+    parser.add_argument(
+        '-f',
+        '--pwdfile',
+        help='Passwords file')
+    parser.add_argument(
+        '-p',
+        '--password',
+        help=('Password for the username. If not specified password will '
+              'be generated'))
+    return parser
+
+
+def check_args(args):
+    if (args.username is None) or (args.pwdfile is None):
+        return False
+
+    return True
+
+
+def generate_password(pwd_len=16):
+    # List of characters to generate password from.
+    # We want to avoid to use similarly looking pairs like
+    # (O, 0), (1, l), etc.
+    lower_case_letters = 'abcdefghijkmnpqrstuvwxyz'
+    upper_case_letters = 'ABCDEFGHIJKLMPQRSTUVWXYZ'
+    digits = '23456789'

```

Comment:
why are 0 and 1 excluded if there is a random include being used (below)

Reply:
Because 0 can be confused with O and 1 can be confused with l.